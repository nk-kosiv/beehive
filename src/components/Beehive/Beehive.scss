.beehive {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 30px;
  width: 100%;

  &__container {
    background-color: rgb(241, 233, 217);
    display: flex;
    justify-content: space-around;
    align-items: center;
    flex-wrap: wrap;
    flex-grow: 1;
    width: 100%;
  }

  &__reset {
    padding: 20px;
    
    &-button {
      padding: 6px 12px;
      background-color: #000;
      color: #fff;
      text-transform: uppercase;
      font-weight: 700;
      border: none;
      border-radius: 3px;

    }
  }
}
