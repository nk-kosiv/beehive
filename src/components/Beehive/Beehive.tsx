import { useState } from "react";
import { Bee } from "../Bee/Bee";
import "./Beehive.scss";

type BeehiveProps = {
  beehive: { health: number }[];
};

export const Beehive: React.FC<BeehiveProps> = ({ beehive }) => {
  const [shouldReset, setShouldReset] = useState(false);

  const resetBeeHealth = () => setShouldReset((prev) => !prev);

  return (
    <div className="beehive">
      <div className="beehive__reset">
        <button className="beehive__reset-button" onClick={resetBeeHealth}>
          Reset
        </button>
      </div>
      <section className="beehive__container">
        {beehive.map(({ health }, beeIndex) => (
          <Bee
            key={beeIndex.toString()}
            health={health}
            shouldReset={shouldReset}
          />
        ))}
      </section>
    </div>
  );
};
