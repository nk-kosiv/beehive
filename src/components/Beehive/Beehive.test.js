/* eslint-disable testing-library/no-unnecessary-act */
/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { render, fireEvent, act } from "@testing-library/react";

import { Beehive } from "./Beehive";

const beehiveMock = [{ health: 100 }, { health: 100 }];

describe("Beehive component related tests", () => {
  it("Should render App", () => {
    const { container } = render(<Beehive beehive={beehiveMock} />);

    expect(container).toMatchSnapshot();
  });

  it("Should render reset bee health", () => {
    const { container } = render(<Beehive beehive={beehiveMock} />);

    const secondBeeMock = container.querySelectorAll(".bee")[1];
    const reduceHealthInput = secondBeeMock.querySelector(
      ".bee__process-input"
    );

    act(() => {
      fireEvent.change(reduceHealthInput, { target: { value: 10 } });
    });
    act(() => {
      fireEvent.keyDown(reduceHealthInput, { key: "Enter" });
    });

    expect(
      secondBeeMock.querySelector(".bee__status-health span").textContent
    ).toBe("90%");

    const resetButton = container.querySelector(".beehive__reset-button");
    act(() => {
      fireEvent.click(resetButton);
    });

    expect(
      secondBeeMock.querySelector(".bee__status-health span").textContent
    ).toBe("100%");
  });
});
