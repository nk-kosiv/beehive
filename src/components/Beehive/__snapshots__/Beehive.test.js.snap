// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beehive component related tests Should render App 1`] = `
<div>
  <div
    class="beehive"
  >
    <div
      class="beehive__reset"
    >
      <button
        class="beehive__reset-button"
      >
        Reset
      </button>
    </div>
    <section
      class="beehive__container"
    >
      <div
        class="bee"
      >
        <img
          alt="bee"
          class="bee__image"
          src="bee.png"
        />
        <div
          class="bee__process"
        >
          <input
            class="bee__process-input"
            type="number"
          />
          <button
            class="bee__process-button"
          >
            Reduce
          </button>
        </div>
        <div
          class="bee__status"
        >
          <div
            class="bee__status-health"
          >
            <span>
              100%
            </span>
          </div>
          <div
            class="bee__status-live"
          >
            <span
              class="bee__status-live-text"
            >
              Alive
            </span>
          </div>
        </div>
      </div>
      <div
        class="bee"
      >
        <img
          alt="bee"
          class="bee__image"
          src="bee.png"
        />
        <div
          class="bee__process"
        >
          <input
            class="bee__process-input"
            type="number"
          />
          <button
            class="bee__process-button"
          >
            Reduce
          </button>
        </div>
        <div
          class="bee__status"
        >
          <div
            class="bee__status-health"
          >
            <span>
              100%
            </span>
          </div>
          <div
            class="bee__status-live"
          >
            <span
              class="bee__status-live-text"
            >
              Alive
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
